src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsKeyProcessorProvider.cs

```diff
@@ -55,85 +53,67 @@ public InlineHintsKeyProcessor(IWpfTextView view)
             {
                 // Disconnect our callbacks.
                 _view.Closed -= OnViewClosed;
                 _view.LostAggregateFocus -= OnLostFocus;
+
+                // Go back to off-mode just so we don't somehow get stuck in on-mode if the option was on when the view closed.
+                ToggleOff();
             }
 
             private void OnLostFocus(object sender, EventArgs e)
             {
                 // if focus is lost (which can happen for shortcuts that include ctrl-alt...) then go back to normal
                 // inline-hint processing.
-                var workspace = GetWorkspace();
-                if (workspace == null)
-                    return;
-
-                ToggleOff(workspace);
+                ToggleOff();
             }
 
             public override void KeyDown(KeyEventArgs args)
             {
                 base.KeyDown(args);
 
-                var workspace = GetWorkspace();
-                if (workspace == null)
-                    return;
-
                 // if this is either the ctrl or alt key, and only ctrl-alt is down, then toggle on. 
                 // otherwise toggle off if anything else is pressed down.
                 if (IsCtrlOrAlt(args))
                 {
                     if (args.KeyboardDevice.Modifiers == (ModifierKeys.Control | ModifierKeys.Alt))
                     {
-                        ToggleOn(workspace);
+                        ToggleOn();
                         return;
                     }
                 }
 
-                ToggleOff(workspace);
+                ToggleOff();
             }
 
             public override void KeyUp(KeyEventArgs args)
             {
                 base.KeyUp(args);
 
-                var workspace = GetWorkspace();
-                if (workspace == null)
-                    return;
-
-                // If we've lifted a key up turn of the inline hints.
-                ToggleOff(workspace);
+                // If we've lifted a key up from ctrl/alt, then turn off the inline hints.
+                if (IsCtrlOrAlt(args))
+                    ToggleOff();
             }
 
-            private static void ToggleOn(Workspace workspace)
-                => Toggle(workspace, on: true);
+            private void ToggleOn()
+                => Toggle(on: true);
 
-            private static void ToggleOff(Workspace workspace)
-                => Toggle(workspace, on: false);
+            private void ToggleOff()
+                => Toggle(on: false);
 
-            private static void Toggle(Workspace workspace, bool on)
+            private void Toggle(bool on)
             {
                 // No need to do anything if we're already in the requested state
-                var state = workspace.Options.GetOption(InlineHintsOptions.DisplayAllOverride);
+                var state = _globalOptionService.GetOption(InlineHintsOptions.DisplayAllOverride);
                 if (state == on)
```

Comment:
Ignore this if you don't agree, but to me this reads confusingly. `desiredState` instead of `on` maybe?

Reply:
That works for me :-)